packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -54,14 +55,15 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
         const resource = await this.resource;
         this.toDispose.push(resource);
         if (resource.onDidChangeContents) {
-            this.toDispose.push(resource.onDidChangeContents(content => this.readPreferences()));
+            this.onDidResourceChanged = resource.onDidChangeContents(() => this.readPreferences());
```

Comment:
I would create a new `toDisposeOnInit` disposable collection and dispose it on `init` call, in subclass i would only call ìnit`.

Reply:
sure I can add `this.resource` and `this.onDidResourceChanged` to a collection named `toDisposeOnWorkspaceLocationChanged`.